### ISSUE TITLE
False positive in fn_to_numeric_cast_with_truncation casting to function pointer

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/rust-clippy#3275]
This lint seems to trigger incorrectly whenever I cast a function to a function pointer. There is no truncation involved here, unless I somehow misunderstand function pointers.

```rust
fn main() {
    fn f() {}
    let _ = f as fn();
}
```

```console
warning: casting function pointer `f` to `fn()`, which truncates the value     
 --> src/main.rs:3:13                                                          
  |                                                                            
3 |     let _ = f as fn();                                                     
  |             ^^^^^^^^^ help: try: `f as usize`                              
  |                                                                            
  = note: #[warn(clippy::fn_to_numeric_cast_with_truncation)] on by default    
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#fn_to_numeric_cast_with_truncation
```

**Comment by phansch** [comment]
Looks like I missed checking the cast target properly, this lint should indeed only care about casts to numerics :+1:
